### PR TITLE
Created the gps_decision node, fixed the gps_manager node, fixed the sim

### LINKED ISSUE #30: GPS_Manager crashes when it reaches the final waypoint

### LINKED ISSUE #119: Robot Gets temporarily semi-stuck in gazebo
- Currently, the robot will quite often get stuck, or at least become slow and unresponsive in gazebo. 
- This likely has something to do with how the physics sorrounding the caster wheels and the joint between the caster wheels and the robot works.

### PR DESCRIPTION
**Pull request by garethellis0**
**Preface**: Sorry about this Val, this PR really should have 3 or 4 separate PR's, but stuff ended up being required from other branches, and I fixed it as I went. Regardless, I'll try not to do this again.

**Major Changes**:
- Wrote the `gps_decision` node
- Added an IMU to the JFrost gazebo model
- Fixed an issue where the robot would get "stuck" (ended up just removing one of the caster wheels from the back)
- Fixed the gps_manager node: Both the tests and the actual node itself were wrong (rotation was inverted :grimacing: )
- Split `LidarObstacle` and `LidarDecision` in separate header and test files

**Misc. Changes**
- Removed `python_catkin_tools` from `install_dependencies.sh`
- Fixed some usages of `abs` that should have been `fabs`
- Cleaned up some stuff in `lidar-decision-test`

**Specific Stuff to Test** (Some testing ideas, alongside overall code review and testing)
- Run the `igvc_qualification.launch` world from `sb_gazebo`, then launch the `gps_manager.launch` and `gps_decision.launch` files from the `decision` package. Robot should navigate to the two gps waypoints (`gps_manager` should print out when it hits one)
- With all the above launch files running, see what the rqt graph looks like and make sure it makes sense
- Look at the `gps_decision.launch` file, and make sure that all the options make sense

closes #119, #44 
closes #30

### CODE REVIEW

File: get_started.sh [hunk 1]
```diff
@@ -29,8 +29,8 @@ declare -a new_shell_config_lines=(
     # Make sure that all shells know where to find our custom gazebo models,
     # plugins, and resources. Make sure to preserve the path that already exists as well
     "export GAZEBO_MODEL_PATH=$DIR/src/sb_gazebo/models:${GAZEBO_MODEL_PATH}"\
-    "export GAZEBO_PLUGIN_PATH=/home/path/to/IGVC-2017/src/sb_gazebo/lib:${GAZEBO_PLUGIN_PATH}"\
-    "export GAZEBO_RESOURCE_PATH=/home/path/to/IGVC-2017/src/sb_gazebo/models:${GAZEBO_RESOURCE_PATH}"\
```
Comment (garethellis0) [inline get_started.sh]: Hmm. Odd these changes would register as changes, seeing as they're already on master? I'm just going to put it down to git being wierd.

Comment (ValRat) [inline get_started.sh]: sp00ky

File: src/decision/include/LidarObstacle.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+ */
+
+#ifndef LIDAROBSTACLE_H
+#define LIDAROBSTACLE_H
+
+// STD
+#include <vector>
+#include <algorithm>
+
+
```
Comment (MathewMacDougall) [inline src/decision/include/LidarObstacle.h]: Extra line here?

File: src/decision/src/GpsDecision.cpp [hunk 3]
```diff
@@ -1,37 +1,65 @@
...
+    SB_getParam(private_nh, "linear_heading_factor", linear_heading_factor, 1.0);
+    SB_getParam(private_nh, "linear_distance_factor", linear_distance_factor, 1.0);
+    SB_getParam(private_nh, "angular_heading_factor", angular_heading_factor, 1.0);
+    SB_getParam(private_nh, "angular_distance_factor", angular_distance_factor, 1.0);
+    mover.setFactors(linear_distance_factor, linear_heading_factor,
+                    angular_distance_factor, angular_heading_factor);
+    SB_getParam(private_nh, "max_linear_speed", linear_cap, 1.0);
+    SB_getParam(private_nh, "max_angular_speed", angular_cap, 1.0);
+    mover.setMaxSpeeds(linear_cap, angular_cap);
+
```
Comment (MathewMacDougall) [inline src/decision/src/GpsDecision.cpp]: extra blank line?

File: src/decision/src/GpsMover.cpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    // The command to return
+    geometry_msgs::Twist command;
+
+    // Set components we don't care about to 0
+    command.linear.y = 0;
+    command.linear.z = 0;
+    command.angular.x= 0;
+    command.angular.y = 0;
+
+
```
Comment (MathewMacDougall) [inline src/decision/src/GpsMover.cpp]: extra blank line?

File: src/decision/src/LidarObstacle.cpp [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
-
 /*
  * Created By: Gareth Ellis
  * Created On: January 22, 2017
- * Description: This class represents a Obstacle, as detected by the lidar
+ * Description: This class represents an Obstacle, as detected by the lidar,
+ *              and provides various utilities for working with obstacles
  */
 
+#include <LidarObstacle.h>
```
Comment (MathewMacDougall) [inline src/decision/src/LidarObstacle.cpp]: doesn't have a heading like other #includes

File: src/decision/src/LidarObstacle.cpp [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
+
```
Comment (MathewMacDougall) [inline src/decision/src/LidarObstacle.cpp]: extra blank line?

File: src/decision/test/gps-mover-test.cpp [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+ */
+geometry_msgs::Point createPoint(double x, double y){
+    geometry_msgs::Point p;
+    p.x = x;
+    p.y = y;
+    p.z = 0;
+    return p;
+}
+
+
```
Comment (MathewMacDougall) [inline src/decision/test/gps-mover-test.cpp]: extra line

File: src/decision/test/gps-mover-test.cpp [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+    EXPECT_DOUBLE_EQ(3.0/4.0 * M_PI, should_be_acute);
+}
+
+//Test where angle are separated by a negative obtuse angle (if just subtracted)
+TEST_F(GpsMoverTest, minAngularChangeTest_negativeObtuseAngle){
+    double should_be_acute = mover1.minAngularChange(-M_PI, M_PI/4);
+    EXPECT_DOUBLE_EQ(-3.0/4.0 * M_PI, should_be_acute);
+}
+
+TEST_F(GpsMoverTest, magicFunctionTest_basic){
```
Comment (MathewMacDougall) [inline src/decision/test/gps-mover-test.cpp]: doesn't have a descriptive comment

**Comment by garethellis0** [comment]
@ValRat - ready for review :grimacing:

File: src/decision/test/gps-mover-test.cpp [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+            current_location,
+            0,
+            waypoint
+            );
+
+    // We should be trying to turn left
+    EXPECT_GE(twist.angular.z, 0);
+}
+
+// Test of the robot facing "backwards" relative to initial heading,
```
Comment (ValRat) [inline src/decision/test/gps-mover-test.cpp]: I missed the line, it's the one right after this

File: src/decision/src/GpsManager.cpp [hunk 10]
```diff
@@ -34,12 +34,12 @@ GpsManager::GpsManager(int argc, char **argv, std::string node_name){
 
     // Setup Subscribers
     uint32_t refresh_rate = 10;
-    std::string raw_gps_topic_name = "/gps_driver/gps";
+    std::string raw_gps_topic_name = "/gps_driver/navsatfix";
     raw_gps_subscriber = public_nh.subscribe(raw_gps_topic_name, refresh_rate,
```
Comment (garethellis0) [inline src/decision/src/GpsManager.cpp]: Good point, set to 1.

File: src/decision/src/GpsManager.cpp [hunk 11]
```diff
@@ -34,12 +34,12 @@ GpsManager::GpsManager(int argc, char **argv, std::string node_name){
...
                                              &GpsManager::rawGpsCallBack, this);
-    std::string compass_topic_name = "/imu/compass";
-    compass_subscriber = public_nh.subscribe(compass_topic_name, refresh_rate,
-                                                &GpsManager::compassCallBack, this);
+    std::string imu_topic_name = "/imu";
+    imu_subscriber = public_nh.subscribe(imu_topic_name, refresh_rate,
+                                             &GpsManager::imuCallback, this);
 
     // Setup Publishers
     uint32_t queue_size = 10;
```
Comment (garethellis0) [inline src/decision/src/GpsManager.cpp]: Also set to 1

File: src/decision/src/GpsManager.cpp (hunk 10 above)
Comment (garethellis0) [inline src/decision/src/GpsManager.cpp]: Yeah, that should be private.... it is actually a private namespace, the variable name is just wrong, see #74 for more details..... in any case, changed.

File: src/decision/test/gps-mover-test.cpp (hunk 9 above)
Comment (garethellis0) [inline src/decision/test/gps-mover-test.cpp]: :man_facepalming: : Changed all instances of obstacle to waypoint.

**Comment by garethellis0** [comment]
@ValRat - fixed everything and replied to all your comments, ready for re-review. Thanks for the commits to! :tada:

**Pull request merged**